internal/apiserver/apiserver.go

```diff
@@ -31,38 +32,38 @@ type GetParamsRequest struct {
 type ApiServer struct {
 	token     string
 	generator *generator.Generator
+	repo      *gitrepo.GitRepo
 }
 
-func New(token string, g *generator.Generator) *ApiServer {
+func New(token string, g *generator.Generator, repo *gitrepo.GitRepo) *ApiServer {
 	return &ApiServer{
 		token:     token,
 		generator: g,
+		repo:      repo,
 	}
 }
 
-func (s *ApiServer) Run() error {
-	r := gin.New()
+func (server *ApiServer) Run() error {
```

Comment:
By convention, receiver names should generally by 1-2 characters. Makes receiver variables easy to differentiate from other variables. 

You also change it here, but not in other functions. So lets just keep it as `s`

From go.dev: https://go.dev/wiki/CodeReviewComments#receiver-names